File: Easy/Python/2315. Count Asterisks.py
class Solution(object):
    def countAsterisks(self, s):
        count = 0
        isBarred = False

        # Look through the string, when a bar is found
        # ignore any asterisks found
        for i in range(len(s)):
            if(not isBarred and s[i] == '|'):
                isBarred = True
            elif(isBarred and s[i] == '|'):
                isBarred = False
            if(not isBarred and s[i] == '*'):
                count += 1
        
        return count

File: Easy/Java/2315. Count Asterisks.java
class Solution {
    public int countAsterisks(String s) {
        int count = 0;
        boolean isBarred = false;

        // Look through the string, when a bar is found
        // ignore any asterisks found
        for(int i = 0; i < s.length(); i++){
            if(!isBarred && s.charAt(i) == '|')
                isBarred = true;
            else if(isBarred && s.charAt(i) == '|')
                isBarred = false;
            if(!isBarred && s.charAt(i) == '*')
                count++;
        }

        return count;
    }
}
